### PR TITLE
ts: Add functionality to subscribe to all program accounts.

### PR DESCRIPTION
**Pull request by xjcaa**
Draft implementation.

#1305 Adds the functionality mentioned in here.

Also our solana web3 dependency is a bit out dated as well!

TODO: Maybe can add context to callback type as slot of update is quite useful if you're indexing data.

### CODE REVIEW

**Comment by armaniferrante** [comment]
I think the slot context information is dropped by the underlying web3.js library. Might need to plumb into that package.

**Comment by xjcaa** [comment]
The `AccountChangeCallback` and `ProgramAccountChangeCallback` both pass back the context. See https://github.com/solana-labs/solana-web3.js/blob/master/src/connection.ts#L1753-L1756

File: ts/src/program/namespace/account.ts [hunk 1]
```diff
@@ -227,6 +230,62 @@ export class AccountClient<
...
+        );
+        ee.emit("change", {
+          publicKey: info.accountId,
+          account,
+        });
+      },
+      commitment,
+      [
+        {
+          memcmp: {
```
Comment (armaniferrante) [inline ts/src/program/namespace/account.ts:258]: This should be `this.coder.memcmp(this._idlAcocunt.name),`. The program client no longer assumes a borsh coder.

**Review by armaniferrante — CHANGES_REQUESTED** [review]
Needs a coder update

**Comment by armaniferrante** [comment]
Closing as stale.

**Pull request closed without merging**